File: sieve.py
import numpy as np
p = 2
foundP = True
r = int(input("What range of prime numbers do you want to find? "))

nums = np.arange(start=1,stop=r+1,step=1)
nums[0] = -1;
while foundP:
	for i in range(2,int(r/p+1)):
		nums[p*i-1] = -1
	foundP = False
	for i in range(p,r):
		if nums[i] != -1:
			p = nums[i]
			foundP = True
			break
for i in nums:
	if i != -1:
		print(i, end=", ")
print()

File: sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
int main(void){
	int range;
	int p = 2;
	bool foundP = true;
	printf("What range of prime numbers do you want to find? ");
	scanf("%d", &range);

	int *nums = malloc(range * sizeof(int));
	for(int i = 0; i < range; i++){
		nums[i] = i+1;
	}

	nums[0] = -1;

	do{
		for(int i = 2; p*i <= range; i++){
			nums[p*i-1] = -1;
		}
		foundP = false;
		for(int i = p; i < range && !foundP; i++){
			if(nums[i] != -1){
				p = nums[i];
				foundP = true;
			}
		}
	}while(foundP);

	for(int i = 0; i < range; i++){
		if(nums[i] != -1){
			printf("%d, ", nums[i]);
		}
	}
	printf("\n");
	free(nums);
	return 0;
}
